src/core/messaging/notifications/notifications.ts

```diff
@@ -1,142 +1,146 @@
-import firebase from 'react-native-firebase';
-import { dataMessageHandler } from '../handlers/data-message';
-import { INotificationPayload, NotificationType } from '../types';
+import messaging from '@react-native-firebase/messaging';
+import { captureException as SentryCaptureException } from '@sentry/react-native';
+import { Platform } from 'react-native';
+import PushNotificationIOS from '@react-native-community/push-notification-ios';
 import { notificationHandler } from '../handlers/notification';
-import { store } from '../../../redux/config';
-import { themes } from '../../../navigation/navigation';
-import { getUnseenNotifications } from '../../../redux/notifications/actions';
+import PushNotification from 'react-native-push-notification';
 
-// this file is in this format for testing purposes
 export class NotificationService {
-    public onTokenRefreshListener = null;
-    public messageListener = null;
-    public notificationDisplayedListener = null;
-    public onNotificationListener = null;
-    public onNotificationOpenedListener = null;
+    private messageListener = null;
+    private messageBackgrounListener = null;
+    private messageNotificationOpenedAppListener = null;
+    private tokenRefreshListener = null;
+    private localNotificationIOSListener = null;
 
-    public async getToken() {
-        return firebase.messaging().getToken();
+    private token: string = null;
+
+    public async configure() {
+        try {
+            const msg = messaging();
+
+            await msg.hasPermission();
+            await msg.requestPermission();
+
+            this.token = await msg.getToken();
+        } catch (error) {
+            SentryCaptureException(new Error(JSON.stringify(error)));
+        }
+
+        this.createListeners();
+
+        this.wasOpenedByNotification();
     }
 
-    public async checkPermission() {
-        // @ts-ignore seems like this is a must for now
-        const enabled = await firebase.messaging().hasPermission();
-        this.requestPermission();
+    public async getToken() {
+        return this.token || messaging().getToken();
```

Comment:
this.token is string - getToken() doesnt return a Promise?

Reply:
yes, but the function is `async` so its result will always be a promise. You can return a `Promise<string>` or directly a `string` both will be fine.